api/registrations/serializers.py

```diff
@@ -74,11 +74,11 @@ class RegistrationSerializer(NodeSerializer):
     )
 
     date_registered = VersionedDateTimeField(source='registered_date', read_only=True, help_text='Date time of registration.')
-    date_withdrawn = VersionedDateTimeField(source='retraction.date_retracted', read_only=True, help_text='Date time of when this registration was retracted.')
+    date_withdrawn = ser.SerializerMethodField(help_text='Date time of when this registration was retracted.')
     embargo_end_date = HideIfWithdrawal(ser.SerializerMethodField(help_text='When the embargo on this registration will be lifted.'))
     custom_citation = HideIfWithdrawal(ser.CharField(allow_blank=True, required=False))
 
-    withdrawal_justification = ser.CharField(source='retraction.justification', read_only=True)
+    withdrawal_justification = ser.SerializerMethodField()
```

Comment:
Instead of making `date_withdrawn` and `withdrawal_justification` SerializerMethodFields on the RegistrationSerializer, I would recommend making them properties on the `Registration` model, which you can use with the VersionedDateTimeField.

Your solution doesn't return the properly formatted date for older versions of the API, which the VersionedDateTimeField takes care of for you.